tests/manage/z_cluster/test_mon_log_trimming.py

```diff
@@ -0,0 +1,191 @@
+from ocs_ci.ocs.resources import pod
+from ocs_ci.ocs.resources.pod import get_mon_pods
+from ocs_ci.utility.utils import convert_device_size
+from ocs_ci.ocs.exceptions import TimeoutExpiredError
+
+log = logging.getLogger(__name__)
+
+
+@tier2
+@bugzilla("1941939")
+@pytest.mark.polarion_id("OCS-2526")
+class TestMonLogTrimming(E2ETest):
+    """
+    Test cases to validate mon store trimming
+
+    Steps:
+      - Change values:
+        paxos_service_trim_min default value = 250
+        paxos_service_trim_max default value = 500
+        osd_op_complaint_time default value = 30.000000
+      - Execute ops that will write to mon store db
+      - Sample the db size and check that it actually get trimmed
+      - Change the above to default values
+    """
+
+    selected_mon_pod = None
+    ct_pod = None
+    initial_db_size = 0
+    mon_db_trim_count = 0
+    current_mon_db_size = 0
+    stop_checking_mon_db = False
+    fio_pod_obj = None
+    MAX_DELTA_DB_SIZE = 0
+    PAXOS_MIN_DEFAULT = 250
+    PAXOS_MAX_DEFAULT = 500
+    PAXOS_COMPLAINT_TIME_DEFAULT = 30.000000
+
+    @pytest.fixture(autouse=True)
+    def setup(self, request, pod_factory):
+        """
+        Set values for:
+          paxos_service_trim_min=10
+          paxos_service_trim_max=100
+          osd_op_complaint_time=0.000001
+        """
+        self.fio_pod_obj = pod_factory(constants.CEPHFILESYSTEM)
+        mon_pods = get_mon_pods()
+        self.selected_mon_pod_obj = random.choice(mon_pods)
+        self.selected_mon_pod = (
+            self.selected_mon_pod_obj.get().get("metadata").get("labels").get("mon")
+        )
+        log.info(f"Selected mon pod is: {self.selected_mon_pod_obj.name}")
+        log.info(
+            "Setting values: paxos_service_trim_min=10, paxos_service_trim_max=100 "
+            "and osd_op_complaint_time=0.000001"
+        )
+        self.ct_pod = pod.get_ceph_tools_pod()
+        # mon in the "tell" command should be mon.a / mon.b / mon.c
+        self.ct_pod.exec_ceph_cmd(
+            ceph_cmd=f"ceph tell mon.{self.selected_mon_pod} injectargs --paxos_service_trim_min=10"
+        )
+        self.ct_pod.exec_ceph_cmd(
+            ceph_cmd=f"ceph tell mon.{self.selected_mon_pod} injectargs --paxos_service_trim_max=100"
+        )
+        self.ct_pod.exec_ceph_cmd(
+            ceph_cmd=f"ceph tell mon.{self.selected_mon_pod} injectargs --osd_op_complaint_time=0.000001"
+        )
+
+        def finalizer():
+            """
+            Set default values for:
+              paxos_service_trim_min=250
+              paxos_service_trim_max=500
+              osd_op_complaint_time=30.000000
+            """
+            if not self.stop_checking_mon_db:
+                self.stop_checking_mon_db = True
+            log.info(
+                "Setting default values for paxos_service_trim_min(250), paxos_service_trim_max(500) "
+                "and osd_op_complaint_time(30.000000)"
+            )
```

Comment:
We can also use the constants' values instead of the numbers: `250`, `500`, and `30.000000`.

Reply:
Done